src/gfnff/gfnff_ini2.f90

```diff
@@ -776,83 +776,107 @@ subroutine gfnff_hbset(n,at,xyz,topo,neigh,nlist,hbthr1,hbthr2)
-            if(iTrDum.le.neigh%numctr.and.iTrDum.gt.0) then
-              ijnonbond=neigh%bpair(j,i,iTrDum).ne.1
-            else
-              ! i and j are not in neighboring cells
-              ijnonbond=.true. 
-            endif
-            ! loop over relevant H atoms
-            do k=1,topo%nathbH
-              free=.true. ! tripplet not assigned yet
-              nh  =topo%hbatHl(1,k) ! nh always in central cell
-              ! distances for non-cov bonded case
-              rih=NORM2(xyz(:,nh)-(xyz(:,i)+neigh%transVec(:,iTri)))**2
-              rjh=NORM2(xyz(:,nh)-(xyz(:,j)+neigh%transVec(:,iTrj)))**2
-              ! check if i is the bonded A    
-              if(iTri.le.neigh%numctr) then ! nh is not shifted so bpair works without adjustment 
-                if(neigh%bpair(i,nh,iTri).eq.1.and.ijnonbond) then
-                  nlist%nhb2=nlist%nhb2+1
-                  nlist%hblist2(1,nlist%nhb2)=i
-                  nlist%hblist2(2,nlist%nhb2)=j
-                  nlist%hblist2(3,nlist%nhb2)=nh
-                  nlist%hblist2(4,nlist%nhb2)=iTri
-                  nlist%hblist2(5,nlist%nhb2)=iTrj
-                  free=.false. ! not available for nhb1 !!!
-                endif 
-              endif  
-              ! check if j is the bonded A
-              if(iTrj.le.neigh%numctr.and.free) then
-                if(neigh%bpair(j,nh,iTrj).eq.1.and.ijnonbond) then
-                  nlist%nhb2=nlist%nhb2+1
-                  nlist%hblist2(1,nlist%nhb2)=j
-                  nlist%hblist2(2,nlist%nhb2)=i
-                  nlist%hblist2(3,nlist%nhb2)=nh
-                  nlist%hblist2(4,nlist%nhb2)=iTrj
-                  nlist%hblist2(5,nlist%nhb2)=iTri
-                  free=.false. ! not available for nhb1 !!!
-                endif
-              endif  
-              ! check for non-cov bonded A  
-              if(rab+rih+rjh.lt.hbthr2.and.free) then ! sum of rAB,rAH,rBH is below threshold
-                nlist%nhb1=nlist%nhb1+1
-                nlist%hblist1(1,nlist%nhb1)=i 
-                nlist%hblist1(2,nlist%nhb1)=j 
-                nlist%hblist1(3,nlist%nhb1)=nh
-                nlist%hblist1(4,nlist%nhb1)=iTri
-                nlist%hblist1(5,nlist%nhb1)=iTrj
-              endif
-            enddo ! k: relevant H atoms
-          enddo ! iTrj
-        enddo ! iTri
+         i=topo%hbatABl(1,ix)
+         j=topo%hbatABl(2,ix)
+         do iTri=1, neigh_nTrans ! go through i shifts
+            do iTrj=1, neigh_nTrans ! go through j shifts
+               ! get adjustet iTr -> for use of neigh% distances and bpair with two shifted atoms
+               iTrDum=neigh%fTrSum(neigh%iTrNeg(iTri),iTrj)
+               ! iTrDum=-1 is valid here because we are only interested if ij is a bond (ijnonbond)
+               ! However, iTrDum is not and should not be used as an index without excluding -1 value
+               if((iTrDum.le.neigh_nTrans.and.iTrDum.gt.0).or.iTrDum.eq.-1) then ! cycle invalid values
```

Comment:
Can you please convert these `if`s to early returns instead to make the code more readable?
I.e., invert the condition and use `continue`.

Reply:
If I am not mistaken, continue is a "do-nothing" statement in Fortran. The previously implemented `cycle` statement is not allowed in OpenMP parallel do regions. Therefore, I changed it to the less readable if statement.

Reply:
Oh yes, I meant `cycle`. However, I do see why it should not be allowed. Actually, we use that rather often ([example](https://github.com/tblite/tblite/blob/d597263248ca5d64fba08accf17234cec5e141b7/src/tblite/repulsion/effective.f90#L159)).

Reply:
You are absolutely right! Somehow my compiler told me I could not use `cycle` at the time, but it seems it has changed its mind about the current version of the code. I reversed it to the original version as requested.